Migrate develop docs to AsciiDoc and under docs/

strogonoff opened the pull request.

dewyatt commented:
It seems odd to have a top-level [`README.adoc`](https://github.com/strogonoff/rnp/blob/master/README.adoc) and also a [`docs/README.adoc`](https://github.com/strogonoff/rnp/blob/master/docs/README.adoc), especially when they start out very similarly.

dewyatt commented on docs/installation.adoc:
```diff
@@ -0,0 +1,55 @@
+----
+# Clone the repository by version tag (or omit it to get the latest sources)
+sudo apt install git
+git clone https://github.com/riboseinc/rnp.git -b v0.11.0
```
We have an 0.12.0 release -- I wonder if we should try to keep this line up-to-date?

ronaldtse replied:
We should definitely keep installation steps up to date, but I wonder if this one in particular we can use some sort of variable...?

strogonoff replied:
How about adding this item to release checklist @dewyatt 
Alongside any changelog updates and such

strogonoff replied:
@dewyatt if no objections, I’ll add this to the release workflow doc (around where it tells the maintainer to update CHANGELOG)

dewyatt replied:
@ronaldtse I'm not aware of any at the moment. @strogonoff That sounds like a good idea to me.

ronaldtse commented:
@strogonoff as @dewyatt mentioned, is the `docs/README.adoc` supposed to take a new life of its own?

strogonoff commented:
> It seems odd to have a top-level [`README.adoc`](https://github.com/strogonoff/rnp/blob/master/README.adoc) and also a [`docs/README.adoc`](https://github.com/strogonoff/rnp/blob/master/docs/README.adoc), especially when they start out very similarly.

@dewyatt that README shouldn’t be there, apparently I moved it and then accidentally recreated by writing from previously open Vim buffer so it didn’t make it in the commit. Deleted now.

strogonoff commented:
To clarity, now the README was moved back into the root. Parts of it were moved in respective files under `docs/` where they will be seen by the project site, and README now links to those.

Sometimes things move back and forth as I’m trying to sort out the best way to handle this, so please don’t mind that.

Thanks for noticing the stale README (I wouldn’t have noticed that until later)

ronaldtse commented:
@strogonoff are we ready to merge?

strogonoff commented:
@ronaldtse Yes, ready as of now.

Sorry, forgetting to add the no-ci tag at commit time

skalee commented:
Perhaps symlinking `README.adoc` to `docs/README.adoc` or _vice versa_ would solve the issue with maintaining two files?

dewyatt commented:
I have some other comments (might be outside the scope of this PR) after looking again:
1. I see we have both a [`doc/tests/`](https://github.com/riboseinc/rnp/tree/63e69a4908eec5b3c41ef25b24464559225dc607/doc/tests) and [`docs/`](https://github.com/riboseinc/rnp/tree/63e69a4908eec5b3c41ef25b24464559225dc607/docs). I know you have an open issue for this (#816) but I still thought it might be good to move it into `docs/` (until we have guidance on whether to remove it entirely). I think it will confuse people.
2. It looks like [`docs/develop.adoc`](https://github.com/riboseinc/rnp/tree/63e69a4908eec5b3c41ef25b24464559225dc607/docs/develop.adoc) would fit nicely in [`docs/develop/`](https://github.com/riboseinc/rnp/tree/63e69a4908eec5b3c41ef25b24464559225dc607/docs/develop). If it were a top-level doc that linked to the items in the `docs/develop` directory, then it would make sense, but it doesn't appear to be that. What do you think?
3. I assume [`docs/navigation.adoc`](https://github.com/riboseinc/rnp/tree/63e69a4908eec5b3c41ef25b24464559225dc607/docs/navigation.adoc) is maybe something for rnpgp.com and not intended to be viewed on GitHub? In that case, I think ideally we would make it appear like less of an end-user file. Mainly I think the name sounds too user-friendly and people will click on it thinking it will be helpful to navigate the docs, but it won't be. Could it be moved to a top-level `.ribose/` or something else?

Also, generally we need to distinguish between "internal development docs" and "development docs" (API docs for end-users). I think having a `docs/internal` might make sense at some point.

strogonoff commented:
> Perhaps symlinking `README.adoc` to `docs/README.adoc` or vice versa would solve the issue with maintaining two files?

@skalee the brief simultaneous existence of README.adoc and docs/README.adoc was an accident, the one under docs/ has been deleted as it should’ve been in the first place.

strogonoff commented:
@dewyatt 

1. If there’re no objections I’ll move test docs under docs/ as well in future

2. https://www.rnpgpg.com/software/rnp/docs/develop/ is intended to be the entry point to further development docs nested under the /develop/ path

3. It’s a valid concern, currently the navigation doc is required by open project site during Jekyll build, in future the navigation may be moved elsewhere or changed so that document contents contain user-clickable navigation.

4. I don’t think there’s any confusion. You have docs about _using_ RNP through its CLI tool and/or C APIs, and about _developing_ RNP itself, and that’s how they’re phrased. Unless you’re closed-source or closed for contributions, I believe calling contribution docs “internal” would be a mistake. It’s great that you have contribution guides, this is not common across Ribose Open projects currently!

I believe the situation with RNP docs will be improved with these changes, if that is so then we can file improvement issues later rather than delay the merge, on the other hand if there’re any show-stoppers then I’d like to hear them!

dewyatt commented:
I think I forgot to merge this!
